### PR TITLE
Add Technisat Dimmer and series switch

### PR DESCRIPTION
**Pull request by fanningert**
Hello, I added a configuration for Technisat dimmer and Technisat series switch. Please merge.

### CODE REVIEW

**Comment by roflmao** [comment]
Need index.json update as well

**Comment by fanningert** [comment]
Thanks. I missed this file. The file is now being adapted.

**Review by AlCalzone — CHANGES_REQUESTED** [review]
Thanks, I found a few things that need to be changed though.

FYI, you can automatically update the index file and check your config files like this:
https://zwave-js.github.io/node-zwave-js/#/development/config-files?id=contributing-configuration-files

File: packages/config/config/devices/0x0299/1a91.json [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+					"value": 0
+				},
+				{
+					"label": "enable central scene",
+					"value": 1
+				}
+			]
+		},
+		"2": {
+			"label": "Parameter 2",
```
[the remark this answers is not in the document]
Comment (fanningert) [reply packages/config/config/devices/0x0299/1a91.json]: It's the same name that the manufacturer are using in the manual.
Sorry, they have only a German manual
https://www.technisat.com/assets/medias/docus/130/BDA_Serienschalter.pdf

File: packages/config/config/devices/0x0299/1a91.json [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+			"valueSize": 2,
+			"minValue": 0,
+			"maxValue": 8640,
+			"defaultValue": 3,
+			"readOnly": false,
+			"writeOnly": false,
+			"allowManualEntry": true
+		},
+		"3": {
+			"label": "Parameter 3",
```
[the remark this answers is not in the document]
Comment (fanningert) [reply packages/config/config/devices/0x0299/1a91.json]: It's the same name that the manufacturer used in the manual.
Sorry, they have only a German manual
https://www.technisat.com/assets/medias/docus/130/BDA_Serienschalter.pdf

File: packages/config/config/devices/0x0299/1a91.json [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+		}
+	],
+	"firmwareVersion": {
+		"min": "0.0",
+		"max": "255.255"
+	},
+	"supportsZWavePlus": true,
+	"paramInformation": {
+		"1": {
+			"label": "Parameter 1",
```
[the remark this answers is not in the document]
Comment (fanningert) [reply packages/config/config/devices/0x0299/1a91.json]: It's the same name that the manufacturer are using in the manual.
Sorry, they have only a German manual
https://www.technisat.com/assets/medias/docus/130/BDA_Serienschalter.pdf

File: packages/config/config/devices/0x0299/1a91.json [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+			"defaultValue": 3,
+			"readOnly": false,
+			"writeOnly": false,
+			"allowManualEntry": true
+		},
+		"3": {
+			"label": "Parameter 3",
+			"description": "Interval of active energy meter reports in minutes. - 10 to 30240 - Interval of active energy meter unsolicited reports in minutes (10 minutes - 3 weeks) - 0 - Disable unsolicited meter reports of active energy.",
+			"valueSize": 2,
+			"minValue": 0,
```
[the remark this answers is not in the document]
Comment (fanningert) [reply packages/config/config/devices/0x0299/1a91.json:58]: Is corrected

File: packages/config/config/devices/0x0299/1a91.json [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+				{
+					"label": "enable central scene",
+					"value": 1
+				}
+			]
+		},
+		"2": {
+			"label": "Parameter 2",
+			"description": "3 to 8640 - Interval of unsolicited reports in 10s increments, a value of 3 means 30 seconds (30 seconds - 1 day) 0 - Disable unsolicited meter reports of current wattage.",
+			"valueSize": 2,
```
[the remark this answers is not in the document]
Comment (fanningert) [reply packages/config/config/devices/0x0299/1a91.json:44]: Is corrected

File: packages/config/config/devices/0x0299/1a92.json [hunk 6]
```diff
@@ -0,0 +1,131 @@
...
+	],
+	"firmwareVersion": {
+		"min": "0.0",
+		"max": "255.255"
+	},
+	"supportsZWavePlus": true,
+	"paramInformation": {
+		"1": {
+			"label": "Parameter 1",
+			"description": "Enable/disable central scene notifications for 2x-5x button presses.",
```
[the remark this answers is not in the document]
Comment (fanningert) [reply packages/config/config/devices/0x0299/1a92.json]: It's the same name that the manufacturer used in the manual.
Sorry, they have only a German manual
https://www.technisat.com/assets/medias/docus/130/BDA_Dimmer.pdf

File: packages/config/config/devices/0x0299/1a92.json [hunk 7]
```diff
@@ -0,0 +1,131 @@
...
+					"value": 0
+				},
+				{
+					"label": "enable central scene",
+					"value": 1
+				}
+			]
+		},
+		"2": {
+			"label": "Parameter 2",
```
[the remark this answers is not in the document]
Comment (fanningert) [reply packages/config/config/devices/0x0299/1a92.json]: It's the same name that the manufacturer used in the manual.
Sorry, they have only a German manual
https://www.technisat.com/assets/medias/docus/130/BDA_Dimmer.pdf

File: packages/config/config/devices/0x0299/1a92.json [hunk 8]
```diff
@@ -0,0 +1,131 @@
...
+			"valueSize": 2,
+			"minValue": 0,
+			"maxValue": 8640,
+			"defaultValue": 3,
+			"readOnly": false,
+			"writeOnly": false,
+			"allowManualEntry": true
+		},
+		"3": {
+			"label": "Parameter 3",
```
[the remark this answers is not in the document]
Comment (fanningert) [reply packages/config/config/devices/0x0299/1a92.json]: It's the same name that the manufacturer are using in the manual.
Sorry, they have only a German manual
https://www.technisat.com/assets/medias/docus/130/BDA_Dimmer.pdf

File: packages/config/config/devices/0x0299/1a92.json [hunk 9]
```diff
@@ -0,0 +1,131 @@
...
+			"valueSize": 2,
+			"minValue": 0,
+			"maxValue": 30240,
+			"defaultValue": 60,
+			"readOnly": false,
+			"writeOnly": false,
+			"allowManualEntry": true
+		},
+		"4": {
+			"label": "Parameter 4",
```
[the remark this answers is not in the document]
Comment (fanningert) [reply packages/config/config/devices/0x0299/1a92.json]: It's the same name that the manufacturer are using in the manual.
Sorry, they have only a German manual
https://www.technisat.com/assets/medias/docus/130/BDA_Dimmer.pdf

File: packages/config/config/devices/0x0299/1a92.json [hunk 10]
```diff
@@ -0,0 +1,131 @@
...
+					"value": 0
+				},
+				{
+					"label": "External Push Button",
+					"value": 1
+				}
+			]
+		},
+		"5": {
+			"label": "Parameter 5",
```
[the remark this answers is not in the document]
Comment (fanningert) [reply packages/config/config/devices/0x0299/1a92.json]: It's the same name that the manufacturer are using in the manual.
Sorry, they have only a German manual
https://www.technisat.com/assets/medias/docus/130/BDA_Dimmer.pdf

File: packages/config/config/devices/0x0299/1a92.json [hunk 11]
```diff
@@ -0,0 +1,131 @@
...
+					"value": 0
+				},
+				{
+					"label": "External Push Button",
+					"value": 1
+				}
+			]
+		},
+		"6": {
+			"label": "Parameter 6",
```
[the remark this answers is not in the document]
Comment (fanningert) [reply packages/config/config/devices/0x0299/1a92.json]: It's the same name that the manufacturer are using in the manual.
Sorry, they have only a German manual
https://www.technisat.com/assets/medias/docus/130/BDA_Dimmer.pdf

**Comment by fanningert** [comment]
@AlCalzone thanks. I will check the link. Sorry for the manuell work. Ich will test ist with the tool.

**EDIT:** I am a beginner with nodeJS. After installation of yarn and ts-node, I get many errors on running "yarn run config import -i". Looks like I need the create the right enviroment for this test. Is there any details for this steps?

Here are my errors.
```bash
$ yarn run config import -i
yarn run v1.22.5
$ ts-node maintenance/importConfig.ts import -i

/home/xxx/github/node-zwave-js/node_modules/ts-node/src/index.ts:513
    return new TSError(diagnosticText, diagnosticCodes)
           ^
TSError: ⨯ Unable to compile TypeScript:
maintenance/importConfig.ts:7:1 - error TS2591: Cannot find name 'process'. Do you need to install type definitions for node? Try `npm i --save-dev @types/node` and then add `node` to the types field in your tsconfig.

7 process.on("unhandledRejection", (r) => {
  ~~~~~~~
maintenance/importConfig.ts:7:35 - error TS7006: Parameter 'r' implicitly has an 'any' type.

7 process.on("unhandledRejection", (r) => {
                                    ~
maintenance/importConfig.ts:11:31 - error TS2307: Cannot find module 'alcalzone-shared/objects' or its corresponding type declarations.

11 import { composeObject } from "alcalzone-shared/objects";
                                 ~~~~~~~~~~~~~~~~~~~~~~~~~~
maintenance/importConfig.ts:12:35 - error TS2307: Cannot find module 'alcalzone-shared/typeguards' or its corresponding type declarations.

12 import { isArray, isObject } from "alcalzone-shared/typeguards";
                                     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
maintenance/importConfig.ts:13:21 - error TS2307: Cannot find module 'ansi-colors' or its corresponding type declarations.

13 import { red } from "ansi-colors";
                       ~~~~~~~~~~~~~
maintenance/importConfig.ts:14:36 - error TS2307: Cannot find module 'assert' or its corresponding type declarations.

14 import { AssertionError, ok } from "assert";
                                      ~~~~~~~~
maintenance/importConfig.ts:15:19 - error TS2307: Cannot find module 'axios' or its corresponding type declarations.

15 import axios from "axios";
                     ~~~~~~~
maintenance/importConfig.ts:16:24 - error TS2307: Cannot find module 'child_process' or its corresponding type declarations.

16 import * as child from "child_process";
                          ~~~~~~~~~~~~~~~
maintenance/importConfig.ts:17:21 - error TS2307: Cannot find module 'fs-extra' or its corresponding type declarations.

17 import * as fs from "fs-extra";
                       ~~~~~~~~~~
maintenance/importConfig.ts:18:24 - error TS2307: Cannot find module 'json5' or its corresponding type declarations.

18 import * as JSON5 from "json5";
                          ~~~~~~~
maintenance/importConfig.ts:19:23 - error TS2307: Cannot find module 'path' or its corresponding type declarations.

19 import * as path from "path";
                         ~~~~~~
maintenance/importConfig.ts:20:25 - error TS2307: Cannot find module 'semver' or its corresponding type declarations.

20 import { compare } from "semver";
                           ~~~~~~~~
maintenance/importConfig.ts:21:27 - error TS2307: Cannot find module 'util' or its corresponding type declarations.

21 import { promisify } from "util";
                             ~~~~~~
maintenance/importConfig.ts:22:22 - error TS2307: Cannot find module 'xml2json' or its corresponding type declarations.

22 import xml2json from "xml2json";
                        ~~~~~~~~~~
maintenance/importConfig.ts:23:19 - error TS2307: Cannot find module 'yargs' or its corresponding type declarations.

23 import yargs from "yargs";
                     ~~~~~~~
maintenance/importConfig.ts:105:2 - error TS2304: Cannot find name '__dirname'.

105  __dirname,
     ~~~~~~~~~
maintenance/importConfig.ts:110:30 - error TS2304: Cannot find name '__dirname'.

110 const ozwTempDir = path.join(__dirname, "../.tmpozw");
                                 ~~~~~~~~~
maintenance/importConfig.ts:116:29 - error TS2304: Cannot find name '__dirname'.

116 const ohTempDir = path.join(__dirname, "../.tmpoh");
                                ~~~~~~~~~
maintenance/importConfig.ts:180:23 - error TS7006: Parameter 'err' implicitly has an 'any' type.

180  	stream.on("error", (err) => {
     	                    ~~~
maintenance/importConfig.ts:730:24 - error TS2532: Object is possibly 'undefined'.

730  if (!isArray(IDs) || !IDs.length) {
                           ~~~
maintenance/importConfig.ts:731:3 - error TS2591: Cannot find name 'process'. Do you need to install type definitions for node? Try `npm i --save-dev @types/node` and then add `node` to the types field in your tsconfig.

731  	process.stdout.write("Fetching database IDs...");
     	~~~~~~~
maintenance/importConfig.ts:734:3 - error TS2591: Cannot find name 'process'. Do you need to install type definitions for node? Try `npm i --save-dev @types/node` and then add `node` to the types field in your tsconfig.

734  	process.stdout.write("\r\x1b[K");
     	~~~~~~~
maintenance/importConfig.ts:738:22 - error TS2532: Object is possibly 'undefined'.

738  for (let i = 0; i < IDs.length; i++) {
                         ~~~
maintenance/importConfig.ts:739:3 - error TS2591: Cannot find name 'process'. Do you need to install type definitions for node? Try `npm i --save-dev @types/node` and then add `node` to the types field in your tsconfig.

739  	process.stdout.write(
     	~~~~~~~
maintenance/importConfig.ts:740:42 - error TS2532: Object is possibly 'undefined'.

740  		`Fetching device config ${i + 1} of ${IDs.length}...`,
     		                                      ~~~
maintenance/importConfig.ts:742:37 - error TS2532: Object is possibly 'undefined'.

742  	const content = await fetchDevice(IDs[i]);
     	                                  ~~~
maintenance/importConfig.ts:744:28 - error TS2532: Object is possibly 'undefined'.

744  		path.join(ohTempDir, `${IDs[i]}.json`),
     		                        ~~~
maintenance/importConfig.ts:749:3 - error TS2591: Cannot find name 'process'. Do you need to install type definitions for node? Try `npm i --save-dev @types/node` and then add `node` to the types field in your tsconfig.

749  	process.stdout.write("\r\x1b[K");
     	~~~~~~~
maintenance/importConfig.ts:756:2 - error TS2591: Cannot find name 'process'. Do you need to install type definitions for node? Try `npm i --save-dev @types/node` and then add `node` to the types field in your tsconfig.

756  process.stdout.write("Fetching manufacturers...");
     ~~~~~~~
maintenance/importConfig.ts:761:2 - error TS2591: Cannot find name 'process'. Do you need to install type definitions for node? Try `npm i --save-dev @types/node` and then add `node` to the types field in your tsconfig.

761  process.stdout.write("\r\x1b[K");
     ~~~~~~~
maintenance/importConfig.ts:891:3 - error TS2591: Cannot find name 'process'. Do you need to install type definitions for node? Try `npm i --save-dev @types/node` and then add `node` to the types field in your tsconfig.

891  	process.exit(1);
     	~~~~~~~
maintenance/importConfig.ts:934:4 - error TS7006: Parameter 'file' implicitly has an 'any' type.

934  	(file) =>
     	 ~~~~
maintenance/importConfig.ts:1104:14 - error TS7006: Parameter 'id' implicitly has an 'any' type.

1104  					?.map((id) => parseInt(id as any))
      					       ~~
maintenance/importConfig.ts:1105:16 - error TS7006: Parameter 'num' implicitly has an 'any' type.

1105  					.filter((num) => !Number.isNaN(num)),
      					         ~~~

    at createTSError (/home/xx/github/node-zwave-js/node_modules/ts-node/src/index.ts:513:12)
    at reportTSError (/home/xxx/github/node-zwave-js/node_modules/ts-node/src/index.ts:517:19)
    at getOutput (/home/xxx/github/node-zwave-js/node_modules/ts-node/src/index.ts:752:36)
    at Object.compile (/home/xxx/github/node-zwave-js/node_modules/ts-node/src/index.ts:968:32)
    at Module.m._compile (/home/xxx/github/node-zwave-js/node_modules/ts-node/src/index.ts:1056:42)
    at Module._extensions..js (internal/modules/cjs/loader.js:1092:10)
    at Object.require.extensions.<computed> [as .ts] (/home/xxx/github/node-zwave-js/node_modules/ts-node/src/index.ts:1059:12)
    at Module.load (internal/modules/cjs/loader.js:928:32)
    at Function.Module._load (internal/modules/cjs/loader.js:769:14)
    at Function.executeUserEntryPoint [as runMain] (internal/modules/run_main.js:72:12)
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

**Comment by AlCalzone** [comment]
> Looks like I need the create the right enviroment for this test

You probably need to execute `yarn` in the local repo to install all missing dependencies.

**Comment by fanningert** [comment]
@AlCalzone Thanks, that's help in the steps. 
```bash
$ sudo dnf install gcc-c++
$ yarn
# Add missing libraries
$ npm install -D typescript
$ npm install -D ts-node
$ npm install -D @zwave-js/core
$ npm install -D @zwave-js/shared
# Retry
$ yarn
$ yarn run config import -i
```

But at the end I also get an error. Which OS version do you use? I am using Fedora 33 (XFCE Spin). Looks like it's not the perfect os for this task. 

Is the current version of this Pull Request correct?

```bash
$ yarn run config import -i
yarn run v1.22.5
$ ts-node maintenance/importConfig.ts import -i
Error: Cannot find module '/home/xxx/github/node-zwave-js/packages/config/node_modules/@zwave-js/core/build/index.js'. Please verify that the package.json has a valid "main" entry
    at tryPackage (internal/modules/cjs/loader.js:303:19)
    at Function.Module._findPath (internal/modules/cjs/loader.js:516:18)
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:867:27)
    at Function.Module._load (internal/modules/cjs/loader.js:725:27)
    at Module.require (internal/modules/cjs/loader.js:952:19)
    at require (internal/modules/cjs/helpers.js:88:18)
    at Object.<anonymous> (/home/xxx/github/node-zwave-js/packages/config/src/DeviceClasses.ts:1:1)
    at Module._compile (internal/modules/cjs/loader.js:1063:30)
    at Module.m._compile (/home/xxx/github/node-zwave-js/node_modules/ts-node/src/index.ts:1056:23)
    at Module._extensions..js (internal/modules/cjs/loader.js:1092:10)
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

**Comment by AlCalzone** [comment]
Just running yarn should install all missing dependencies. Manually installing them will mess up the local symlinks, especially this:

    # Add missing libraries
    $ npm install -D typescript
    $ npm install -D ts-node
    $ npm install -D @zwave-js/core
    $ npm install -D @zwave-js/shared

To clean up, run

    rm -rf node_modules && rm -rf packages/*/node_modules && yarn

**Comment by fanningert** [comment]
Sorry, but I get an error at the end.

```bash
$ git clone ...
$ cd node-zwave-js
$ rm -rf node_modules && yarn 
yarn install v1.22.5
[1/4] Resolving packages...
[2/4] Fetching packages...
info fsevents@2.1.3: The platform "linux" is incompatible with this module.
info "fsevents@2.1.3" is an optional dependency and failed compatibility check. Excluding it from installation.
info fsevents@1.2.13: The platform "linux" is incompatible with this module.
info "fsevents@1.2.13" is an optional dependency and failed compatibility check. Excluding it from installation.
info fsevents@2.2.1: The platform "linux" is incompatible with this module.
info "fsevents@2.2.1" is an optional dependency and failed compatibility check. Excluding it from installation.
[3/4] Linking dependencies...
warning " > @xstate/test@0.4.1" has unmet peer dependency "xstate@^4.9.0".
warning "@xstate/test > @xstate/graph@1.3.0" has unmet peer dependency "xstate@^4.9.0".
warning "lerna > @lerna/version > @lerna/github-client > @octokit/rest > @octokit/plugin-request-log@1.0.2" has unmet peer dependency "@octokit/core@>=3".
[4/4] Building fresh packages...
Done in 24.83s.
$ yarn run config import -i
yarn run v1.22.5
$ ts-node maintenance/importConfig.ts import -i

/home/xxx/github/node-zwave-js/node_modules/ts-node/src/index.ts:513
    return new TSError(diagnosticText, diagnosticCodes)
           ^
TSError: ⨯ Unable to compile TypeScript:
packages/config/src/DeviceClasses.ts:1:61 - error TS2307: Cannot find module '@zwave-js/core' or its corresponding type declarations.

1 import { CommandClasses, ZWaveError, ZWaveErrorCodes } from "@zwave-js/core";
                                                              ~~~~~~~~~~~~~~~~
packages/config/src/DeviceClasses.ts:2:37 - error TS2307: Cannot find module '@zwave-js/shared' or its corresponding type declarations.

2 import { JSONObject, num2hex } from "@zwave-js/shared";
                                      ~~~~~~~~~~~~~~~~~~
packages/config/src/DeviceClasses.ts:100:4 - error TS2571: Object is of type 'unknown'.

100  		e.code === ZWaveErrorCodes.Config_Invalid
     		~
packages/config/src/DeviceClasses.ts:105:39 - error TS2571: Object is of type 'unknown'.

105  				`Could not load scales config: ${e.message}`,
     				                                 ~

    at createTSError (/home/xxx/github/node-zwave-js/node_modules/ts-node/src/index.ts:513:12)
    at reportTSError (/home/xxx/github/node-zwave-js/node_modules/ts-node/src/index.ts:517:19)
    at getOutput (/home/xxx/github/node-zwave-js/node_modules/ts-node/src/index.ts:752:36)
    at Object.compile (/home/xxx/github/node-zwave-js/node_modules/ts-node/src/index.ts:968:32)
    at Module.m._compile (/home/xxx/github/node-zwave-js/node_modules/ts-node/src/index.ts:1056:42)
    at Module._extensions..js (internal/modules/cjs/loader.js:1092:10)
    at Object.require.extensions.<computed> [as .ts] (/home/xxx/github/node-zwave-js/node_modules/ts-node/src/index.ts:1059:12)
    at Module.load (internal/modules/cjs/loader.js:928:32)
    at Function.Module._load (internal/modules/cjs/loader.js:769:14)
    at Module.require (internal/modules/cjs/loader.js:952:19)
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

**Comment by AlCalzone** [comment]
> I get an error at the end.

Weird, it works for me:

```
domin@DESKTOP-OCULUR0 MINGW64 /c/Repositories/node-zwave-js (pr/fanningert/1252)
$ yarn
yarn install v1.22.5
[1/4] Resolving packages...
[2/4] Fetching packages...
info fsevents@2.1.3: The platform "win32" is incompatible with this module.
info "fsevents@2.1.3" is an optional dependency and failed compatibility check. Excluding it from installation.
info fsevents@1.2.13: The platform "win32" is incompatible with this module.
info "fsevents@1.2.13" is an optional dependency and failed compatibility check. Excluding it from installation.
info fsevents@2.2.1: The platform "win32" is incompatible with this module.
info "fsevents@2.2.1" is an optional dependency and failed compatibility check. Excluding it from installation.
[3/4] Linking dependencies...
warning " > @xstate/test@0.4.1" has unmet peer dependency "xstate@^4.9.0".
warning "@xstate/test > @xstate/graph@1.3.0" has unmet peer dependency "xstate@^4.9.0".
warning "lerna > @lerna/version > @lerna/github-client > @octokit/rest > @octokit/plugin-request-log@1.0.2" has unmet peer dependency "@octokit/core@>=3".
[4/4] Building fresh packages...
Done in 13.07s.

domin@DESKTOP-OCULUR0 MINGW64 /c/Repositories/node-zwave-js (pr/fanningert/1252)
$ yarn run config import -i
yarn run v1.22.5
$ ts-node maintenance/importConfig.ts import -i
Done in 3.73s.
```

I'm on Windows, but also develop in a debian container. Maybe you need to run `yarn run build` once before the import script.

**Review by AlCalzone — APPROVED** [review]

**Pull request merged**